src/System.Net.Http/tests/UnitTests/HPack/HPackDecoderTest.cs

```diff
@@ -0,0 +1,583 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Net.Http.HPack;
+using Xunit;
+using System.Buffers;
+
+namespace System.Net.Http.Unit.Tests.HPack
+{
+    public class HPackDecoderTest
+    {
+        private const int DynamicTableInitialMaxSize = 4096;
+
+        // Indexed Header Field Representation - Static Table - Index 2 (:method: GET)
+        private static readonly byte[] _indexedHeaderStatic = new byte[] { 0x82 };
+
+        // Indexed Header Field Representation - Dynamic Table - Index 62 (first index in dynamic table)
+        private static readonly byte[] _indexedHeaderDynamic = new byte[] { 0xbe };
+
+        // Literal Header Field with Incremental Indexing Representation - New Name
+        private static readonly byte[] _literalHeaderFieldWithIndexingNewName = new byte[] { 0x40 };
+
+        // Literal Header Field with Incremental Indexing Representation - Indexed Name - Index 58 (user-agent)
+        private static readonly byte[] _literalHeaderFieldWithIndexingIndexedName = new byte[] { 0x7a };
+
+        // Literal Header Field without Indexing Representation - New Name
+        private static readonly byte[] _literalHeaderFieldWithoutIndexingNewName = new byte[] { 0x00 };
+
+        // Literal Header Field without Indexing Representation - Indexed Name - Index 58 (user-agent)
+        private static readonly byte[] _literalHeaderFieldWithoutIndexingIndexedName = new byte[] { 0x0f, 0x2b };
+
+        // Literal Header Field Never Indexed Representation - New Name
+        private static readonly byte[] _literalHeaderFieldNeverIndexedNewName = new byte[] { 0x10 };
+
+        // Literal Header Field Never Indexed Representation - Indexed Name - Index 58 (user-agent)
+        private static readonly byte[] _literalHeaderFieldNeverIndexedIndexedName = new byte[] { 0x1f, 0x2b };
+
+        private const string _userAgentString = "user-agent";
```

Comment:
Nit: we generally name all consts with PascalCasing.  Same for other consts below.

Reply:
I agree, but this file is largely copied from ASP.NET's repo with only minor changes to support slightly different API surface areas. I'm hesitant to make styling changes as it will make diffs in the future more difficult. What do you think?

Ultimately I'd like to setup automatic mirroring of the entire HPACK code across the two repos, but that's for another time.

Reply:
What file is it a copy of?

Reply:
https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/test/HPackDecoderTests.cs

Reply:
Ok. Can you please then at least add a comment to the top of the file explaining what this is a copy of?

Reply:
Also, do we need to do something special with regards to license? You removed that file's Apache license header. cc: @richlander

Reply:
That's a good call out. The entire HPACK code itself originated from ASP.NET and those other files had their licenses removed as well.

Reply:
Updated licensing to original and added TPN per @richlander